src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Http/HeaderDictionaryAddFixer.cs

```diff
@@ -0,0 +1,150 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Analyzers.Http.Fixers;
+
+[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+public class HeaderDictionaryAddFixer : CodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticDescriptors.DoNotUseIHeaderDictionaryAdd.Id);
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            var appendTitle = "Use IHeaderDictionary.Append";
+            context.RegisterCodeFix(
+                CodeAction.Create(appendTitle,
+                    cancellationToken => ReplaceAddWithAppendAsync(diagnostic, context.Document, cancellationToken),
+                    equivalenceKey: appendTitle),
+                diagnostic);
+
+            var indexerTitle = "Use the indexer";
+            context.RegisterCodeFix(
+                CodeAction.Create(indexerTitle,
+                    cancellationToken => ReplaceAddWithIndexerAsync(diagnostic, context.Document, cancellationToken),
+                    equivalenceKey: indexerTitle),
+                diagnostic);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> ReplaceAddWithAppendAsync(Diagnostic diagnostic, Document document, CancellationToken cancellationToken)
+    {
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        if (root is not CompilationUnitSyntax compilationUnitSyntax)
+        {
+            return document;
+        }
+
+        var invocation = compilationUnitSyntax.FindNode(diagnostic.Location.SourceSpan);
+
+        if (invocation is InvocationExpressionSyntax { Expression: MemberAccessExpressionSyntax { Name.Identifier: { } identifierToken } })
+        {
+            compilationUnitSyntax = compilationUnitSyntax.ReplaceToken(identifierToken, SyntaxFactory.Identifier("Append"));
```

Comment:
I'm not sure, but could you check if `SyntaxFactory.Identifier("Append").WithAdditionalAnnotations(Simplifier.AddImportsAnnotation)` will avoid the need to manually add the `using`?

Reply:
Hmm, I just tried this but it doesn't seem to add the `using`. Is there anything else that I would need to do here besides adding the call to `WithAdditionalAnnotations(Simplifier.AddImportsAnnotation)` on the identifier?

Reply:
@david-acker Try getting the symbol for `Microsoft.AspNetCore.Http.HeaderDictionaryExtensions` and create an annotation like the following:

```csharp
var annotation = new SyntaxAnnotation("SymbolId", DocumentationCommentId.CreateReferenceId(symbol));
```

Then use `SyntaxFactory.Identifier("Append").WithAdditionalAnnotations(Simplifier.AddImportsAnnotation, annotation)`